Common/Data/Market/QuoteBar.cs

```diff
@@ -0,0 +1,420 @@
+        {
+            var midpoint = (bidPrice + askPrice) / 2;
+            var midhigh = Math.Max(lastTrade, midpoint);
+            var midlow = Math.Min(lastTrade, midpoint);
+
+            //Assumed not set yet. Will fail for custom time series where "price" $0 is a possibility.
+            if (Open == 0) Open = lastTrade;
+            if (Bid.Open == 0) Bid.Open = bidPrice;
+            if (Ask.Open == 0) Ask.Open = askPrice;
+
+            if (midhigh > High) High = midhigh;
+            if (bidPrice > Bid.High) Bid.High = bidPrice;
+            if (askPrice > Ask.High) Ask.High = askPrice;
+
+            if (midlow < Low) Low = midlow;
+            if (bidPrice < Bid.Low) Bid.Low = bidPrice;
+            if (askPrice < Ask.Low) Ask.Low = askPrice;
+
+            // Note: we are summing instead of averaging!!!
+            AvgBidSize += Convert.ToInt32(bidSize);
+            AvgAskSize += Convert.ToInt32(askSize);
+
+            //Always set the closing price;
+            Close = lastTrade;
+            Bid.Close = bidPrice;
+            Ask.Close = askPrice;
+            Value = lastTrade;
+        }
+
+        /// <summary>
+        /// QuoteBar Reader: Fetch the data from the QC storage and feed it line by line into the engine.
+        /// </summary>
+        /// <param name="config">Symbols, Resolution, DataType, </param>
+        /// <param name="line">Line from the data file requested</param>
+        /// <param name="date">Date of this reader request</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>Enumerable iterator for returning each line of the required data.</returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            //Handle end of file:
+            if (line == null)
+            {
+                return null;
+            }
+
+            if (isLiveMode)
+            {
+                return new QuoteBar();
+            }
+
+            try
+            {
+                switch (config.SecurityType)
+                {
+                    //Equity File Data Format:
+                    case SecurityType.Equity:
+                        return ParseEquity<QuoteBar>(config, line, date);
+
+                    //FOREX has a different data file format:
+                    case SecurityType.Forex:
+                        return ParseForex<QuoteBar>(config, line, date);
+                }
+            }
+            catch (Exception err)
+            {
+                Log.Error("DataModels: QuoteBar(): Error Initializing - " + config.SecurityType + " - " + err.Message + " - " + line);
+            }
+
+            // if we couldn't parse it above return a default instance
+            return new QuoteBar { Symbol = config.Symbol, Period = config.Increment };
```

Comment:
Typical behavior is to return null when we were unable to parse, why return this new instance?